app/models/manageiq/providers/azure/cloud_manager/provision/state_machine.rb

```diff
@@ -6,4 +6,22 @@ def customize_destination
 
     signal :post_create_destination
   end
+
+  def cleanup_instances
+    return if phase_context[:clone_options].nil?
+
+    delete_instance(phase_context[:clone_options])
+  end
+
+  def delete_instance(instance_attrs)
+    source.with_provider_connection do |azure|
+      nis = ::Azure::Armrest::Network::NetworkInterfaceService.new(azure)
+      ips = ::Azure::Armrest::Network::IpAddressService.new(azure)
+      ni = nis.get(instance_attrs[:name], resource_group.name)
+      ip = ips.get("#{instance_attrs[:name]}-publicIp", resource_group.name)
+      sleep(180)
```

Comment:
Wonder if there's a way we can leverage the queue and say *run this 3 minutes from now"